### PR TITLE
Fix data sync testing to use a tmp registry dir

### PR DESCRIPTION
**Pull request by mooneyme**
## Description
This PR should overwrite pushing the local registry to dsgrid-test-data repo and when approved will eliminate the need for this PR: https://github.com/dsgrid/dsgrid-test-data/pull/7.

## Checklist

- [x] Tests exercising the new feature or bug fix
- [x] All tests pass
- [ ] At least one code review approval
- [x] Consider transferring TODOs to JIRA or GitHub

### CODE REVIEW

**Comment by mooneyme** [comment]
@daniel-thom 
The tests are now failing because I moved the `tests/make_us_data_registry.py` to `dsgrid/tests/make_us_data_registry.py` because I needed to import it into `tests_aws/test_registry/test_data_sync.py`. 

`make_us_data_registry.py` relies on some test data in `tests/data/registry` currently. Shouldn't this be moved to the dsgrid-test-data submodule instead? That way we can import it from there and move the `make_us_data_registry.py` to `dsgrid/tests`?

**Comment by daniel-thom** [comment]
> @daniel-thom The tests are now failing because I moved the `tests/make_us_data_registry.py` to `dsgrid/tests/make_us_data_registry.py` because I needed to import it into `tests_aws/test_registry/test_data_sync.py`.
> 
> `make_us_data_registry.py` relies on some test data in `tests/data/registry` currently. Shouldn't this be moved to the dsgrid-test-data submodule instead? That way we can import it from there and move the `make_us_data_registry.py` to `dsgrid/tests`?

pytest automatically runs the code in tests/conftest.py every time you run a pytest command. Code in that that file ensures that there is a up-to-date registry in `tests/data/registry`. It runs `python tests/make_us_data_registry.py` if that directory doesn't exist or is stale because of a new commit. You should be able to fix the tests by fixing the path to this file on line 37.

File: tests_aws/test_registry/test_data_sync.py [hunk 1]
```diff
@@ -26,222 +78,251 @@ def s3_cloudinterface():
...
-        )
-
-        project_id = "efs_2018"
-        dataset_id = None
-        registry_manager.data_sync(project_id, dataset_id)
-
-        assert (
-            len(
-                LocalFilesystem().listdir(local_registry_data_sync / "data", directories_only=True)
+def test_data_sync_project_id(s3_cloudinterface, local_registry):
```
Comment (daniel-thom) [inline tests_aws/test_registry/test_data_sync.py:81]: Is it necessary for both the test and the `s3_cloudinterface` fixture to run the `local_registry` fixture?

Comment (mooneyme) [reply tests_aws/test_registry/test_data_sync.py:81]: I can do that, I just didn't want to repeat that code for every function.

**Comment by mooneyme** [comment]
@daniel-thom tests are passing. Ready to merge?

**Review by daniel-thom — APPROVED** [review]

**Pull request merged**